src/schema-form.tsx

```diff
@@ -93,31 +75,16 @@ export const SchemaForm = ({
         }
     }
 
-    const isObject = (item: any) => {
-        return item && typeof item === 'object' && !Array.isArray(item)
-    }
-
-    const addProperties = (currentSchema: any, newProperties: any): any => {
-        if (isObject(currentSchema) && isObject(newProperties)) {
-            for (const key in newProperties) {
-                if (isObject(newProperties[key])) {
-                    if (!currentSchema[key]) {
-                        Object.assign(currentSchema, { [key]: {} })
-                    }
-                    addProperties(currentSchema[key], newProperties[key])
-                } else {
-                    Object.assign(currentSchema, { [key]: newProperties[key] })
-                }
+    const removeProperties = (currentSchema: any, baseSchema: any, nestedPath: string) => {
+        if (isObject(currentSchema) && isObject(baseSchema)) {
+            if (currentSchema['$ref'] !== undefined) {
+                const def = currentSchema['$ref'].substr(currentSchema['$ref'].lastIndexOf('/') + 1)
+                addProperties(currentSchema, root!.definitions[def])
```

Comment:
what happens if definitions is not defined on root? (I assume that addProperties will be executed adding another call on the stack) - if yes can we do a check before executing `addProperties`